### PR TITLE
Bunny drill example

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- We had been thinking about making a bunny drill example for a while. Here it is. It is a bit more light hearted than the other examples.

# Description of the solution

- The bunny drill is here. It's a chill example. It works quite well actually.

# How Has This Been Tested?

- Ran the drill

# Documentation

- Documented the drill

# Future changes

- Short examples like this one are also a nice way to show features. I like it.

### CODE REVIEW

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 1]
```diff
@@ -0,0 +1,207 @@
+==================================
+Bunny Drill
+==================================
+
+This example simulates the drilling motion of a bunny within a bed of particles. It illustrates how the DEM module of Lethe can simulate complex moving objects and is a testament to our love of lagomorphs. Do not worry friend, no bunnies were hurt in the making of this example!
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:5]:
```suggestion
This example simulates the drilling motion of a bunny within a bed of particles. It illustrates that the DEM module of Lethe can simulate complex moving objects and is a testament to our love of lagomorphs. Do not worry friend, no bunnies were hurt in the making of this example!
```

Comment (voferreira) [reply doc/source/examples/dem/bunny-drill/bunny-drill.rst:5]: Thank god. I was worried about the bunny... oouff

Comment (voferreira) [reply doc/source/examples/dem/bunny-drill/bunny-drill.rst:5]: The disclaimer is so important I would make a .. warning:: out of it.

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    set type               = cylinder
+    set grid type          = balanced
+    set grid arguments     = 6 : 0.10 : 0.25
+    set initial refinement = 2
+  end
+
+Insertion Info
+~~~~~~~~~~~~~~~~~~~
+
+An insertion box is defined inside and the top part of the cylinder. The insertion region is sufficiently high to ensure that there are no collisiosn between the particles bouncing on the bottom wall (or on particles accumulated at the bottom) and the ones being inserted.
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:49]:
```suggestion
An insertion box is defined at the top part of the cylinder. The insertion region is sufficiently high to ensure that there are no collisions between the particles bouncing on the bottom wall (or on particles accumulated at the bottom) and the ones being inserted.
```

Comment (voferreira) [reply doc/source/examples/dem/bunny-drill/bunny-drill.rst:49]: "guarantee that particles are not inserted at another particle's position" or "avoid overlapping at insertion"?

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    set insertion first direction                      = 0
+    set insertion second direction                     = 2
+    set insertion third direction                      = 1
+  end
+
+
+Lagrangian Physical Properties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The total number of particles in this simulation is 8000. All particles have a diameter of 10 mm. The particles are relatively stiff with a Young Modulus of :math:`E=10^7` to ensure that the overlap between the bunny drill and the particles does not become sufficiently large to allow particles to *jump through* the bunny.
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:75]:
```suggestion
The total number of particles in this simulation is 8000. All particles have a diameter of 10 mm. The particles are relatively stiff, with a Young Modulus of :math:`E=10^7` to ensure that the overlap between the bunny drill and the particles does not become sufficiently large to allow particles to *jump through* the bunny.
```

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    set log frequency     = 1000
+    set output frequency  = 1000
+    set output path       = ./output/
+    set output boundaries = true
+  end
+
+Simulation Control (Drilling)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The time end of the simulation is 4.75 seconds after which the drill has done one back-and-forth drilling motion.
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:123]:
```suggestion
The time end of the simulation is 4.75 seconds after which the bunny has done one back-and-forth drilling motion.
```

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    set output path       = ./output/
+    set output boundaries = true
+  end
+
+
+
+Solid Objects (Drilling)
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The bunny is defined using the solid objects feature of Lethe. The surface mesh of the bunny is a GMSH file. The translational of the velocity is defined to have a periodic motion of the motion in along the axis of the cylinder and the bunny is  rotating at a constant angular velocity. This complex drilling motion is fully parametrized from the input file using the function parser functionallities of the translational and the angular velocity of the solid object.
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:141]:
```suggestion
The bunny is defined using the solid objects feature of Lethe. The surface mesh of the bunny is a GMSH file. The translational velocity is defined to have a periodic motion of the motion in along the axis of the cylinder and the bunny is rotating at a constant angular velocity. This complex drilling motion is fully parametrized from the input file using the function parser functionalities of the translational and the angular velocity of the solid object.
```

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+----------------------
+Running the Simulation
+----------------------
+The loading can be simulated using:
+
+.. code-block:: text
+
+  mpirun -np 8 dem bunny-drill-loading.prm
+
+Whereas the drilling is launched after the loading using:
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:175]: Simulation time?

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst (hunk 3 above)
Comment (voferreira) [reply doc/source/examples/dem/bunny-drill/bunny-drill.rst:75]: Do you mean the bunny's skin is not as soft and fluffy as a real bunny's? :/

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+
+.. code-block:: text
+
+  mpirun -np 8 dem bunny-drill.prm
+
+
+-------
+Results
+-------
+As seen in the following two animations, the bunny drills into the particles which generates a complex motion within the granular matter. There is not much more to say here, it is a bunny drill.
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:185]: You mean, no experimental data in the literature to validate or nothing?

File: doc/source/examples/dem/bunny-drill/bunny-drill.rst [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+.. raw:: html
+
+    <iframe width="500" height="600" src="https://www.youtube.com/embed/VcJ_nt9iNmA" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+----------------------------
+Possibilities for Extension
+----------------------------
+
+- Use finer particles to see if the drilling dynamics if affected by the particle size
+- Use an STL of an alternative animal. Although we believe lagomorphs are amazing, we are also fans of mustelidae (e.g. otters) and chinchillidae (e.g. chinchillas or, even better, viscachas). Feel free to replace the drill with your favorite animal and to send us your animation to lethe.cfd@gmail.com
```
Comment (voferreira) [inline doc/source/examples/dem/bunny-drill/bunny-drill.rst:204]:
```suggestion
- Use an STL of an alternative animal. Although we believe lagomorphs are amazing, we are also fans of mustelidae (e.g. otters) and chinchillidae (e.g. chinchillas or, even better, viscachas). Feel free to replace the drill with your favorite animal and send us your animation to lethe.cfd@gmail.com
```

File: examples/dem/3d-bunny-drill/bunny-drill-loading.prm [hunk 9]
```diff
@@ -0,0 +1,112 @@
+# Listing of Parameters
+# ---------------------
```
Comment (voferreira) [inline examples/dem/3d-bunny-drill/bunny-drill-loading.prm:2]: Can you remove the spaces between `#` and `-`? 👀

Comment (voferreira) [inline examples/dem/3d-bunny-drill/bunny-drill-loading.prm:2]:
```suggestion
#----------------------
```

File: examples/dem/3d-bunny-drill/bunny-drill-loading.prm [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+set dimension = 3
+
+# --------------------------------------------------
```
Comment (voferreira) [inline examples/dem/3d-bunny-drill/bunny-drill-loading.prm:6]:
```suggestion
#---------------------------------------------------
```

File: examples/dem/3d-bunny-drill/bunny-drill-loading.prm [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+subsection simulation control
+  set time step         = 2e-6
+  set time end          = 2
+  set log frequency     = 1000
+  set output frequency  = 1000
+  set output path       = ./output/
+  set output boundaries = true
+end
+
+# --------------------------------------------------
```
Comment (voferreira) [inline examples/dem/3d-bunny-drill/bunny-drill-loading.prm:19]:
```suggestion
#---------------------------------------------------
```

File: examples/dem/3d-bunny-drill/bunny-drill.prm [hunk 12]
```diff
@@ -0,0 +1,118 @@
+# Listing of Parameters
+# ---------------------
```
Comment (voferreira) [inline examples/dem/3d-bunny-drill/bunny-drill.prm:2]:
```suggestion
#----------------------
```

**Review by voferreira — CHANGES_REQUESTED** [review]
Loved it! The next step is the cow fluid dynamics!!

**Review by tonielgeitani — APPROVED** [review]
I like this example. It is fun to see.

**Comment by blaisb** [comment]
@lpsaavedra I need your OK to be able to merge :)

**Review by lpsaavedra — APPROVED** [review]

**Pull request merged**